refactor(cdk/testing): remove deprecated protractor harness environment from public API (#24304)
angular/components: src/cdk/testing/protractor/README.md

### No longer publicly available

> This code is no longer publicly available. The protractor harness envrionment became
> deprecated in v12 and became unavailable with Angular CDK v14.

Reviewer: ```suggestion
The protractor harness environment is no longer part of the public API. It was
deprecated in v12 and removed from the API in v14.
```

Author: oh yeah, and also a typo! my english today 😶‍🌫️
